### PR TITLE
Bulk upload validations merge with api

### PR DESCRIPTION
**Pull request by wozzer72**
This is a fairly large PR for review against https://trello.com/c/FUtQEhVR.

It covers extending the bulk upload specific validation to incorporate the entity property framework validation.

To do this I have introduced the `EntityValidator` base class to all four entities: Establishment, Worker, Qualification and Training, which intercepts the existing `hasMandatoryProperties` and `isValid` entity validators.

Establishments/Workers are based on Change Properties. Training/Qualifications are based on internal properties. Consequently, the way they valid is slightly different.

I've had to reduce the severity in response for both Training and Qualification when there is a validation error;l before it raised an exception, now it returns false (aligned with Workers/Establishment).

I also had to fix an error I introduced to `Establishment::toJSON` in respect to the created/updated/updatedBy properties; this is bombing the integration tests.

With the entities now raising `ValidationMessages`, `bulkupload.js` validation functions are now checking for entity validity before caching the entity, and if not valid, merging the validation errors and warnings, with those raised during CSV parsing/validation - including mapping the line number of the associated CSV line validator.

Finally, the `[POST] validate` has been updated to filter between errors and warning and sort on line number.

If you want to gander at the resulting validation intermediary data files, have a look here: https://s3.console.aws.amazon.com/s3/buckets/sfcbulkuploadfiles/479/validation/?region=eu-west-1&tab=overview.

### CODE REVIEW

**Comment by wozzer72** [comment]
[(5) Bulk Upload - Validate Files - Back End](https://trello.com/c/FUtQEhVR/58-5-bulk-upload-validate-files-back-end)

**Comment by wozzer72** [comment]
[(5) Bulk Upload - Validate Files - Back End](https://trello.com/c/FUtQEhVR/58-5-bulk-upload-validate-files-back-end)

**Comment by wozzer72** [comment]
[(5) Bulk Upload - Validate Files - Back End](https://trello.com/c/FUtQEhVR/58-5-bulk-upload-validate-files-back-end)

File: server/models/classes/establishment.js [hunk 1]
```diff
@@ -27,8 +30,10 @@ const SEQUELIZE_DOCUMENT_TYPE = require('./user/userProperties').SEQUELIZE_DOCUM
 // WDF Calculator
 const WdfCalculator = require('./wdfCalculator').WdfCalculator;
 
-class Establishment {
+class Establishment extends EntityValidator {
```
Comment (AndrewKeig) [inline server/models/classes/establishment.js]: Nice, I was going to suggest this.

**Pull request merged**